src/components/Canary/table.js

```diff
@@ -1,149 +1,314 @@
-                    return -1;
-                  }
-                  return 0;
-                })
-                .map((group) => (
-                  <TableGroupRow
-                    showIcon
-                    key={group}
-                    title={group}
-                    items={checks[group]}
-                    onClick={onClick}
-                  />
-                ))}
-            </>
-          ) : (
-            <>
-              {checks.map((check) => (
-                <tr
-                  key={check.key}
-                  onClick={() => onClick(check)}
-                  className="cursor-pointer"
-                >
-                  <td className="px-6 py-2 w-full max-w-0 overflow-hidden overflow-ellipsis whitespace-nowrap">
-                    <Title key={`${check.key}-title`} check={check} />
-                  </td>
-                  <td className="px-6 py-2 whitespace-nowrap">
-                    <StatusList key={`${check.key}-status`} check={check} />
-                  </td>
-                  <td className="px-6 py-2 whitespace-nowrap">
-                    <Uptime key={`${check.key}-uptime`} check={check} />
-                  </td>
-                  <td className="px-6 py-2 whitespace-nowrap">
-                    <Latency key={`${check.key}-latency`} check={check} />
-                  </td>
-                </tr>
-              ))}
-            </>
-          )}
-        </tbody>
-      </table>
-    </div>
+    <Table
+      data={data}
+      columns={columns}
+      labels={labels}
+      history={history}
+      onUnexpandableRowClick={onCheckClick}
+      hasGrouping={groupBy !== "no-group"}
+      {...rest}
+    />
   );
 }
 
-function TableGroupRow({ title, items, onClick, showIcon, ...rest }) {
-  const [expanded, setExpanded] = useState(false);
-  const [aggregated, setAggregated] = useState(null);
+export function Table({
+  data,
+  columns,
+  labels,
+  history,
+  hasGrouping = false,
+  onUnexpandableRowClick,
+  ...rest
+}) {
+  const {
+    state: tableState,
+    getTableProps,
+    getTableBodyProps,
+    headerGroups,
+    rows,
+    setSortBy,
+    prepareRow,
+    toggleHideColumn,
+    toggleRowExpanded
+  } = useTable(
+    { columns, data, disableMultiSort: true, autoResetSortBy: false },
+    useSortBy,
+    useExpanded
+  );
+
+  const tableSortState = tableState?.sortBy;
+
+  // Hide expander column if there is no grouping
+  useEffect(() => {
+    toggleHideColumn("expander", !hasGrouping);
+  }, [hasGrouping, toggleHideColumn]);
+
+  // Set table's sort state according to url params on page load or url change
+  useEffect(() => {
+    const searchParams = window.location.search;
+    const decodedParams = decodeUrlSearchParams(searchParams);
+    if (decodedParams.sortBy) {
+      setSortBy([{ id: decodedParams.sortBy, desc: decodedParams.sortDesc }]);
+    } else {
+      setSortBy([{ id: columnsKeyed.name.accessor, desc: false }]);
```

Comment:
Default values need to be built in every render, so you need to initialise with defaults and then overwrite if they are present.